### ISSUE TITLE
Calling PyObject_SetAttrString with v equal to NULL causes a seg fault

### ISSUE THREAD

**Issue by juliannguyen4** [pypy/pypy#4997]
This doesn't match CPython's behavior: https://docs.python.org/3/c-api/object.html#c.PyObject_SetAttrString

Workaround:

I replaced all calls for PyObject_SetAttrString with this function:
```C
inline void PyObject_SetAttrStringSafe(PyObject *obj, const char *attr_name,
                                       PyObject *value)
{
    if (value == NULL) {
        PyObject_DelAttrString(obj, attr_name);
    }
    else {
        PyObject_SetAttrString(obj, attr_name, value);
    }
}
```
This fixes the seg fault that occurs when `value` is `NULL`. But the expected behavior is `PyObject_SetAttrString` should not seg fault.

**Comment by mattip** [comment]
Right, the code does not support `value==NULL`. In our defense, the documentation does state that the use is deprecated:
> If `v` is NULL, the attribute is deleted. This behaviour is deprecated in favour of using `PyObject_DelAttr`, but there are currently no plans to remove it.

**Comment by mattip** [comment]
I added tests in d4390816768. They pass both untranslated and after translation on py3.9. What version of PyPy are you using?

**Comment by mattip** [comment]
Ahh, but by the time you are calling `EXCEPTION_ON_ERROR` there is already an error set, correct? I seem to remember that is risky: all bets are off when calling a C-API function when there is already an error.

**Comment by juliannguyen4** [comment]
The error indicator in Python is being set after the line where the seg fault happens: https://github.com/aerospike/aerospike-client-python/blob/cf35901c75e7655735b37d17872c13d7bcc61ce7/src/main/client/operate.c#L86

**Comment by mattip** [comment]
But a PyErr could already have been set by any of the code in `pyobject_to_key`. I think you need to call `PyErr_Clear` as the first step in that macro. See [exception handling](https://docs.python.org/3/c-api/exceptions.html):
> If the error is not handled or carefully propagated, additional calls into the Python/C API may not behave as intended and may fail in mysterious ways
